.github/workflows/cbmc-update.yml

```diff
@@ -0,0 +1,79 @@
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+name: Attempt CBMC update
+
+on:
+  schedule:
+    - cron: "30 4 * * Mon" # Run this every Monday at 04:30 UTC
+  workflow_dispatch:     # Allow manual dispatching for a custom branch / tag.
+
+permissions:
+  checks: write
+  contents: write
+  issues: write
+  pull-requests: write
+
+jobs:
+  create-cargo-update-pr:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Kani
+        uses: actions/checkout@v3
+
+      - name: Setup Kani Dependencies
+        uses: ./.github/actions/setup
+        with:
+          os: ubuntu-22.04
+
+      - name: Compare CBMC versions
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          grep ^CBMC_VERSION kani-dependencies >> $GITHUB_ENV
+          CBMC_LATEST=$(gh -R diffblue/cbmc release list | grep Latest | awk '{print $1}' | cut -f2 -d-)
+          echo "CBMC_LATEST=$CBMC_LATEST" >> $GITHUB_ENV
+          if [ x$CBMC_LATEST = x$CBMC_VERSION ] ; then
+            echo "next_step=none" >> $GITHUB_ENV
+          elif gh issue list -S \
+              "CBMC upgrade to $CBMC_LATEST failed" \
+              --json number,title | grep title ; then
+            echo "next_step=none" >> $GITHUB_ENV
+          elif ! git ls-remote --exit-code origin cbmc-$CBMC_LATEST ; then
+            CBMC_LATEST_MAJOR=$(echo $CBMC_LATEST | cut -f1 -d.)
+            CBMC_LATEST_MINOR=$(echo $CBMC_LATEST | cut -f2 -d.)
+            sed -i "s/^CBMC_MAJOR=.*/CBMC_MAJOR=\"$CBMC_MAJOR\"/" kani-dependencies
+            sed -i "s/^CBMC_MINOR=.*/CBMC_MINOR=\"$CBMC_MINOR\"/" kani-dependencies
+            sed -i "s/^CBMC_VERSION=.*/CBMC_VERSION=\"$CBMC_LATEST\"/" kani-dependencies
+            git diff
+            if ! cargo build-dev ; then
+              echo "next_step=create_issue" >> $GITHUB_ENV
+            elif ! ./scripts/kani-regression.sh ; then
+              echo "next_step=create_issue" >> $GITHUB_ENV
```

Comment:
Why do we have two cases here if the result is the same and the regression script calls `cargo build-dev`?